tests/connection/connection.cpp

```diff
@@ -379,6 +379,104 @@ TEST(Connection, twoValidConnections) {
+        "<map_variables variable_1=\"variable1\" variable_2=\"variable2\"/>"
+      "</connection>"
+      "<connection>"
+        "<map_components component_1=\"component1\" component_2=\"component3\"/>"
+        "<map_variables variable_1=\"variable1\" variable_2=\"variable3\"/>"
+      "</connection>"
+      "<connection>"
+        "<map_components component_1=\"component2\" component_2=\"component3\"/>"
+        "<map_variables variable_1=\"variable2\" variable_2=\"variable3\"/>"
+      "</connection>"
+    "</model>";
+    const std::string e2 =
+    "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+    "<model xmlns=\"http://www.cellml.org/cellml/1.2#\">"
+      "<component name=\"component1\">"
+        "<variable name=\"variable1\"/>"
+      "</component>"
+      "<component name=\"component2\">"
+        "<variable name=\"variable2\"/>"
+      "</component>"
+      "<component name=\"component3\">"
+        "<variable name=\"variable3\"/>"
+      "</component>"
+      "<connection>"
+        "<map_components component_1=\"component1\" component_2=\"component2\"/>"
+        "<map_variables variable_1=\"variable1\" variable_2=\"variable2\"/>"
+      "</connection>"
+      "<connection>"
+        "<map_components component_1=\"component1\" component_2=\"component3\"/>"
+        "<map_variables variable_1=\"variable1\" variable_2=\"variable3\"/>"
+      "</connection>"
+    "</model>";
+    const std::string e3 =
+    "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+    "<model xmlns=\"http://www.cellml.org/cellml/1.2#\">"
+      "<component name=\"component1\">"
+        "<variable name=\"variable1\"/>"
+      "</component>"
+      "<component name=\"component2\">"
+        "<variable name=\"variable2\"/>"
+      "</component>"
+      "<component name=\"component3\">"
+        "<variable name=\"variable3\"/>"
+      "</component>"
+    "</model>";
+
+    libcellml::Model m;
+    libcellml::ComponentPtr comp1 = std::make_shared<libcellml::Component>();
+    libcellml::ComponentPtr comp2 = std::make_shared<libcellml::Component>();
+    libcellml::ComponentPtr comp3 = std::make_shared<libcellml::Component>();
+    libcellml::VariablePtr v1 = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v2 = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v3 = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v4 = std::make_shared<libcellml::Variable>();
+    comp1->setName("component1");
+    comp2->setName("component2");
+    comp3->setName("component3");
+    v1->setName("variable1");
+    v2->setName("variable2");
+    v3->setName("variable3");
+
+    comp1->addVariable(v1);
+    comp2->addVariable(v2);
+    comp3->addVariable(v3);
+    m.addComponent(comp1);
+    m.addComponent(comp2);
+    m.addComponent(comp3);
+    libcellml::Variable::addEquivalence(v1,v2);
+    libcellml::Variable::addEquivalence(v1,v3);
+    libcellml::Variable::addEquivalence(v2,v3);
```

Comment:
Eh, I did put "Nitpicking" in front, hence this is just for information and this is in no way meant to prevent a PR merge. :) (As it happens, I have merged PRs for which I had made some nitpicky comments. :))